libs/providers/unleash-web/src/lib/unleash-web-provider.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  EvaluationContext,
+  Provider,
+  Logger,
+  JsonValue,
+  FlagNotFoundError,
+  OpenFeatureEventEmitter,
+  ProviderEvents,
+  ResolutionDetails,
+  ProviderFatalError,
+  TypeMismatchError,
+} from '@openfeature/web-sdk';
+import { UnleashClient } from 'unleash-proxy-client';
+import { UnleashConfig } from './unleash-web-provider-config';
+
+export class UnleashWebProvider implements Provider {
+  metadata = {
+    name: UnleashWebProvider.name,
+  };
+
+  public readonly events = new OpenFeatureEventEmitter();
+
+  // logger is the OpenFeature logger to use
+  private _logger?: Logger;
+
+  // config is the Unleash config provided to the provider
+  private _config?: UnleashConfig;
+
+  // client is the Unleash client reference
+  private _client?: UnleashClient;
+
+  readonly runsOn = 'client';
+
+  constructor(config: UnleashConfig, logger?: Logger) {
+    this._config = config;
+    this._logger = logger;
+    this._client = new UnleashClient(config);
+  }
+
+  public get unleashClient() {
+    return this._client;
+  }
+
+  async initialize(): Promise<void> {
+    await this.initializeClient();
+    this._logger?.info('UnleashWebProvider initialized');
```

Comment:
It may be better to use debug into of info since this is a library but it's up to you.

Reply:
no worries i'll change it so it logs at debug level